include/swift/AST/DiagnosticEngine.h

```diff
@@ -679,6 +693,12 @@ namespace swift {
     /// Path to diagnostic documentation directory.
     std::string diagnosticDocumentationPath = "";
 
+    /// Diagnostic messages language code.
+    std::string diagnosticLocaleCode = "";
```

Comment:
I removed them from `YAMLDiagnosticProvider` as you can see. 

Do you suggest adding the `diagnosticLocaleCode` and `diagnosticMessagesPath` only in `YAMLDiagnosticProvider` and remove them from `DiagnosticEngine`?

Reply:
What I'm trying to say is that both locale and path are associated with localization producer so they should be attached to it instead of diagnostic engine, if they ever need to be accessed anywhere else, or otherwise they shouldn't even be fields of localization provider and only used once.